FirebaseVertexAI/Sources/VertexAI.swift

```diff
@@ -156,25 +146,25 @@ public class VertexAI {
   /// Lock to manage access to the `instances` array to avoid race conditions.
   private static var instancesLock: os_unfair_lock = .init()
 
-  let projectID: String
-  let apiKey: String
   let location: String
 
   init(app: FirebaseApp, location: String) {
-    self.app = app
-    appCheck = ComponentType<AppCheckInterop>.instance(for: AppCheckInterop.self, in: app.container)
-    auth = ComponentType<AuthInterop>.instance(for: AuthInterop.self, in: app.container)
-
     guard let projectID = app.options.projectID else {
       fatalError("The Firebase app named \"\(app.name)\" has no project ID in its configuration.")
     }
-    self.projectID = projectID
-
     guard let apiKey = app.options.apiKey else {
       fatalError("The Firebase app named \"\(app.name)\" has no API key in its configuration.")
     }
-    self.apiKey = apiKey
-
+    firebaseInfo = FirebaseInfo(
+      appCheck: ComponentType<AppCheckInterop>.instance(
+        for: AppCheckInterop.self,
+        in: app.container
+      ),
+      auth: ComponentType<AuthInterop>.instance(for: AuthInterop.self, in: app.container),
+      projectID: projectID,
+      apiKey: apiKey,
+      googleAppID: app.isDataCollectionDefaultEnabled ? app.options.googleAppID : nil
```

Comment:
Could you check if the `isDataCollectionDefaultEnabled` is already sent via another header? If not, I'd prefer a separate header for `isDataCollectionDefaultEnabled` instead of bundling these two because now we can't distinguish between app ID not found and data collection disabled.

Reply:
Wouldn't we end up collecting the app ID if we do so? Or is it ok to collect if it gets discarded? My understanding is that the client should be responsible for managing the data collection flag. cc: @ryanwilson @ncooke3

Reply:
Yes, normally, the client should be responsible for managing data collection. However, in this case, clients can't send data directly to Cloud Monitoring/Logging/Trace. It's done on the proxy server. How can the server know when **not** to send telemetry data? A dedicated header would be cleaner than piggy backing on the presence of AppId. Would this work?
1. A separate header for whether isDataCollectionDefaultEnabled
2. `X-Firebase-AppId` is null/unset if `isDataCollectionDefaultEnabled = false`?

Reply:
Discussed offline and agreed to keep the code as is - only adding `X-Firebase-AppId` to the header for now.

Reply:
Can we change to read the flag at send time? Right now if someone configures Vertex, then disables the global flag, it wouldn't be honoured. I recognize that's more complex, and maybe a refactor / notification system for the global flag makes sense...

Reply:
That's the wrong direction for our APIs. We should head towards making FIRApp immutable to support Swift 6 and structured concurrency. I'd much rather document to set the flag before initializing VertexAI.

Reply:
Let's chat about it - I think we need to be enable flows like an end user going to a settings screen in an app and toggling data collection, and the developer being able to update that flag without tearing down and re-configuring Firebase (which we don't support right now). 

Another flow: have it off by default, then in the user onboarding have an opt in screen that the developer can opt-in on.

Reply:
With this implementation, reconfiguring Firebase wouldn't be necessary, only recreating another VertexAI instance from the updated Firebase App.

Reply:
> With this implementation, reconfiguring Firebase wouldn't be necessary, only recreating another VertexAI instance from the updated Firebase App.

Sorry if I'm missing something obvious in the flow. The `VertexAI` instances are cached in the `static var instances: [String: VertexAI]` above so I think the only way  a dev could recreate a `VertexAI` instance would be to configure a new `FirebaseApp` with a different name to change the setting.

Reply:
You're right Andrew. Disregard my previous comment.

Reply:
Updated to test the latest data collection flag value during the request in latest commit - https://github.com/firebase/firebase-ios-sdk/pull/14479/commits/b189b5c96294ed9f4af16ed4359977038278dd42